compiler/src/iree/compiler/Codegen/SPIRV/SPIRVLinkExecutables.cpp

```diff
@@ -40,36 +42,37 @@ bool operator<(const ExecutableTargetAttr &a, const ExecutableTargetAttr &b) {
   }
   return aFeatures.size() < bFeatures.size();
 }
-} // namespace IREE::HAL
 
 namespace {
 
-using IREE::HAL::ExecutableTargetAttr;
+// Returns all executables that have one or more variants that use SPIR-V
+// codegen. Executables that contain object references are currently ignored as
+// we only support full replacement of the modules and not yet linking.
+static SmallVector<IREE::HAL::ExecutableOp>
+gatherExecutablesForSPIRVCodegen(mlir::ModuleOp moduleOp) {
+  SmallVector<IREE::HAL::ExecutableOp> result;
+  for (auto executableOp : moduleOp.getOps<IREE::HAL::ExecutableOp>()) {
+    if (llvm::count_if(executableOp.getOps<IREE::HAL::ExecutableVariantOp>(),
```

Comment:
`llvm:any_of`?

Reply:
nice! I was hoping for something like that - tried finding an if_range or whatnot